rs/bitcoin/kyt/src/lib.rs

```diff
@@ -56,26 +58,32 @@ impl FetchEnv for KytCanisterEnv {
         state::FetchGuard::new(txid)
     }
 
+    fn btc_network(&self) -> BtcNetwork {
+        get_config().btc_network
+    }
+
     async fn http_get_tx(
         &self,
+        provider: providers::Provider,
         txid: Txid,
         max_response_bytes: u32,
     ) -> Result<Transaction, HttpGetTxError> {
         // TODO(XC-159): Support multiple providers
-        let request = providers::create_request(get_config().btc_network, txid, max_response_bytes);
+        let request = provider.create_request(txid, max_response_bytes);
         let url = request.url.clone();
         let cycles = get_tx_cycle_cost(max_response_bytes);
         match http_request(request, cycles).await {
             Ok((response,)) => {
                 // Ensure response is 200 before decoding
                 if response.status != 200u32 {
-                    let code = if response.status == 429u32 {
+                    let code = if response.status == 429u32 || response.status == 500u32 {
```

Comment:
Those some some good reasons to treat non `200` status as transient. I think for 404s we may want to return `TransactionNotFound` back to the caller as a retriable error, just to make it prominent. What do you think?

Reply:
Do you want to return `TransactionNotFound` after just one call or after trying all providers?
We have to avoid the scenario where the "default provider" lags behind, often returning this error, and then the same provider is used again when the user tries again a few seconds later.

Reply:
> We have to avoid the scenario where the "default provider" lags behind, often returning this error, and then the same provider is used again when the user tries again a few seconds later.

Yes, exactly. I see 2 possible solutions:

1. We only return `TransactionNotFound` after having tried all providers as suggested by Thomas.
2. If we return `TransactionNotFound` directly, the canister would need to remember that for the next time the same transaction is queried, another provider must be used.

In both cases I'm not sure having a dedicated `TransactionNotFound` error help, e.g. think of the following unlucky scenario:

1. Provider 1 lags behind -> 404
2. Provider 2 is offline -> 500
3. Provider 3 rate-limit -> 429

With solution 1 we would need to somehow aggregate errors from multiple providers. With solution 2 we would not need to do that, but having a dedicated error just for transaction not found does seem a bit weird. 

With that in mind, I would treat all errors from providers the same (non 200 status code or a 200 with incorrect data). Does-it make sense?